### DISCUSSION TITLE
how to type a useMutation fetch function?

### DISCUSSION THREAD

**Discussion by hiwyn** [TanStack/query#4531] [General] [answered]
I'm having problems to type a fetch function from a useMutation

```ts
const queryClient = useQueryClient();
  const { mutate: mutateCloseStatus } = useMutation(
    (id: number, newValue: 0 | 1) => changeClosedStatus(id, newValue),
    {
      onSuccess: () => {
        queryClient.invalidateQueries(`topic:${topicId}`);
      },
    }
  );
  ```
  
  ```ts
  export async function changeClosedStatus(topicId: number, newValue: 0 | 1) {
  const response = await api.put(`forum/topic/close/${topicId}`, {
    closedStatus: newValue,
  });
  return response;
}
  ```
  
  returning this error:
  ![alt text](https://media.discordapp.net/attachments/102860784329052160/1043631369164685332/image.png)
  
  what should i do?

**Comment by TkDodo** [comment, marked as answer]
a mutation can only accept _one_ parameter, not many. if you need multiple parameters, pass an object.

https://tkdodo.eu/blog/mastering-mutations-in-react-query#mutations-only-take-one-argument-for-variables

**Reply by hiwyn** [reply]
hmm, thank you
